app/src/main/java/org/cru/godtools/ui/tools/ToolsAdapter.kt

```diff
@@ -51,13 +59,15 @@ class ToolsAdapter(lifecycleOwner: LifecycleOwner, private val dataModel: ToolsA
         tools = t
     }
 
-    override fun getItemViewType(position: Int) = when (getItem(position)?.type) {
-        Tool.Type.LESSON -> VIEW_TYPE_LESSON
+    override fun getItemViewType(position: Int) = when {
+        getItem(position)?.type == Tool.Type.LESSON -> VIEW_TYPE_LESSON
+        itemLayout == R.layout.dashboard_list_item_tools -> VIEW_TYPE_ALL_TOOL
         else -> VIEW_TYPE_TOOL
     }
 
     override fun onCreateViewDataBinding(parent: ViewGroup, viewType: Int) = when (viewType) {
         VIEW_TYPE_LESSON -> ToolsListItemLessonBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        VIEW_TYPE_ALL_TOOL -> DashboardListItemToolsBinding.inflate(LayoutInflater.from(parent.context), parent, false)
```

Comment:
because we want to be able to support any custom layout, you shouldn't use this method to inflate custom layouts, use `DataBindingUtil` instead.